apps/backend/core/configuration/validation/orchestrator.py

```
@@ -0,0 +1,467 @@
+                # Run with fail-fast: stop on first CRITICAL failure
+                for coro in asyncio.as_completed(validation_tasks):
+                    result = await coro
+                    results.append(result)
+
+                    # Check for CRITICAL failure
+                    if (
+                        result.status == ValidationStatus.FAIL
+                        and result.level == ValidationLevel.CRITICAL
+                    ):
+                        logger.error(
+                            f"CRITICAL failure detected: {result.service_name} - {result.error_message}"
```

Ruff (0.14.4)

Logging statement uses f-string

(G004)